### PR TITLE
Static Analysis: Rewrite  empty and inappropriate Javadocs

### PR DESCRIPTION
**Pull request by felixYyu**
fix #4376 cc @RussellSpitzer

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/BaseFilesTable.java [hunk 1]
```diff
@@ -108,7 +108,8 @@ public TableScan appendsAfter(long fromSnapshotId) {
     }
 
     /**
-     * @return list of manifest files to explore for this files metadata table scan
+     * Returns list of {@link ManifestFile} files to explore for this file's metadata table scan
+     * @return list of manifest files
```
Comment (rdblue) [inline core/src/main/java/org/apache/iceberg/BaseFilesTable.java:112]: Why is this change needed?

File: hive3/src/main/java/org/apache/hadoop/hive/ql/io/orc/OrcSplit.java [hunk 2]
```diff
@@ -191,12 +191,12 @@ public boolean hasFooter() {
   }
 
   /**
-   * @return {@code true} if file schema doesn't have Acid metadata columns
    * Such file may be in a delta_x_y/ or base_x due to being added via
    * "load data" command.  It could be at partition|table root due to table having
    * been converted from non-acid to acid table.  It could even be something like
    * "warehouse/t/HIVE_UNION_SUBDIR_15/000000_0" if it was written by an
    * "insert into t select ... from A union all select ... from B"
+   * @return {@code true} if file schema doesn't have Acid metadata columns
```
Comment (rdblue) [inline hive3/src/main/java/org/apache/hadoop/hive/ql/io/orc/OrcSplit.java:199]: I think this can be improved, but moving this line to the end doesn't help readability. Can you rewrite this doc instead?

File: core/src/main/java/org/apache/iceberg/BaseFilesTable.java (hunk 1 above)
Comment (felixYyu) [reply core/src/main/java/org/apache/iceberg/BaseFilesTable.java:112]: the static check warning log ，so I think should need change:

```
[MissingSummary] A summary fragment is required; consider using the value of the @return block as a summary fragment instead.
     * @return list of manifest files to explore for this files metadata table scan
       ^
    (see https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment)
  Did you mean '*Returns list of manifest files to explore for this files metadata table scan.'?
```

File: hive3/src/main/java/org/apache/hadoop/hive/ql/io/orc/OrcSplit.java (hunk 2 above)
Comment (felixYyu) [reply hive3/src/main/java/org/apache/hadoop/hive/ql/io/orc/OrcSplit.java:198]: removing redundant e.g.

File: core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java [hunk 3]
```diff
@@ -27,9 +27,6 @@
 import org.apache.iceberg.relocated.com.google.common.collect.Maps;
 import org.apache.iceberg.rest.RESTResponse;
 
-/**
- *
- */
```
Comment (rdblue) [inline core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java:32]: I think the correct fix is to add javadoc, not remove the placeholder.

File: hive3/src/main/java/org/apache/hadoop/hive/ql/io/orc/OrcSplit.java (hunk 2 above)
Comment (rdblue) [reply hive3/src/main/java/org/apache/hadoop/hive/ql/io/orc/OrcSplit.java:198]: I think the right fix is to document the method, not to just move the `@return`. Can you fix the documentation here instead? Javadoc should start with a short (one sentence) description and then go into more detail if needed after a paragraph break.

Comment (felixYyu) [reply hive3/src/main/java/org/apache/hadoop/hive/ql/io/orc/OrcSplit.java:196]: thanks, I got it and tried to description it.

File: core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java (hunk 3 above)
Comment (felixYyu) [reply core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java:32]: added javadoc

**Comment by felixYyu** [comment]
thanks very much @RussellSpitzer @rdblue suggests, Help me to impove my skills.

**Comment by felixYyu** [comment]
Can you take a look if have time? @rdblue

File: core/src/main/java/org/apache/iceberg/BaseFilesTable.java [hunk 4]
```diff
@@ -115,7 +115,7 @@ protected BaseAllFilesTableScan(TableOperations ops, Table table, Schema schema,
     }
 
     /**
-     * Returns an iterable of manifest files to explore for this all files metadata table scan
+     * Returns list of {@link ManifestFile} files to explore for this file's metadata table scan
```
Comment (rdblue) [inline core/src/main/java/org/apache/iceberg/BaseFilesTable.java:118]: This isn't correct. Can you revert these changes?

File: hive3/src/main/java/org/apache/hadoop/hive/ql/io/orc/OrcSplit.java [hunk 5]
```diff
@@ -191,12 +191,15 @@ public boolean hasFooter() {
...
-   * been converted from non-acid to acid table.  It could even be something like
+   * A check for whether or not this file was originally created in a Hive ACID table.
+   *
+   * A file may be in a delta_x_y/ or base_x due to being added via
+   * "load data" command.  It could be located in a partition|table root due to table having
+   * been converted from non-acid to acid table.  It could also even have paths containing
    * "warehouse/t/HIVE_UNION_SUBDIR_15/000000_0" if it was written by an
    * "insert into t select ... from A union all select ... from B"
+   *
+   * @return true if the file was written originally by a Hive ACID table
```
Comment (rdblue) [inline hive3/src/main/java/org/apache/hadoop/hive/ql/io/orc/OrcSplit.java:202]: There is no need to change this `@return`. The new version is more confusing because it is up to the Javadoc reader to know what it means to be originally created in a Hive ACID table. Instead, just use the original, "true if the file schema does not have ACID metadata columns". That should also be the first sentence description.

In addition, Javadoc paragraphs should be separated by `<p>` on its own line.

Comment (felixYyu) [reply hive3/src/main/java/org/apache/hadoop/hive/ql/io/orc/OrcSplit.java]: thanks, fixed

File: core/src/main/java/org/apache/iceberg/BaseFilesTable.java (hunk 4 above)
Comment (felixYyu) [reply core/src/main/java/org/apache/iceberg/BaseFilesTable.java]: reverted
